runtime/src/iree/modules/vmvx/module.c

```diff
@@ -636,8 +638,9 @@ IREE_VMVX_ABI_EXPORT(iree_vmvx_pack, pack, v) {
                            /*stride1=*/1,
```

Comment:
should this be `args->out_stride1`?

Reply:
(Addressing this comment and the next) - Very good question, and actually that is how I initially wrote it, and tests were mysteriously failing. What was happening was an early-return case hidden in those `MAP_BUFFER_2D_UNTYPED_RO`, `MAP_BUFFER_2D_UNTYPED_RW` macros:
https://github.com/iree-org/iree/blob/5145f7e4e9ab65de1dd91a015440526af678b826/runtime/src/iree/modules/vmvx/module.c#L154-L158
What was happening is that this check was failing in `iree_vm_buffer_map`:
https://github.com/iree-org/iree/blob/5145f7e4e9ab65de1dd91a015440526af678b826/runtime/src/iree/vm/buffer.c#L26

What is going on here is that we are abusing shared macros to map 2D buffers (`MAP_BUFFER_2D_UNTYPED_RO`, `MAP_BUFFER_2D_UNTYPED_RW`) to map 4D buffers. We do that by taking advantage of the fact that the inner 3 dimensions are contiguous, so we coalesce them by treating these 3 dimensions as a single dimension with product size:
https://github.com/iree-org/iree/blob/5145f7e4e9ab65de1dd91a015440526af678b826/runtime/src/iree/modules/vmvx/module.c#L610
and
https://github.com/iree-org/iree/blob/5145f7e4e9ab65de1dd91a015440526af678b826/runtime/src/iree/modules/vmvx/module.c#L638

The value 1 for the stride there then simply reflects the fact that this is now treated as the inner-most dimension.

It's a ugly hack, but it's inside vmvx so... probably not worth fixing :-)